### ISSUE TITLE
Overriding a static path

### ISSUE THREAD

**Issue by avsm** [rgrinberg/opium#25]
I'd like to have a default middleware that attempts to find a static file if no other dynamic override is present.  E.g.:

```
  |> App.middleware (Middleware.static ~local_path:"../html" ~uri_prefix:"/")
  |> Dynamic.generate_index task
```

In this case, the `/` Uri prefix is immediately served as a `Not_found` -- it looks like the `Dynamic` registration for `/` is being ignored despite it being passed to the next handler by the `static` middleware.  Is this intentional?  I'm working around it by specifically registering multiple static servers for `images/` `css/` etc in my site.

**Comment by rgrinberg** [comment]
This quirk of the current implementation was meant to save some cycles but I can see why this might not be desirable. I think pass through definitely makes more sense however.

I've pushed an initial fix for this in master. Will test it more tomorrow.

**Comment by rgrinberg** [comment]
Unfortunately my fix isn't good enough as it makes it impossible to override a path that's provided by the static middleware.

Will look into this further. The real fix is of course to make middleware mounting a little smarter (e.g. mount middleware on a path)

**Comment by avsm** [comment]
Makes sense; the workaround is also relatively easy and probably more sensible (just expose the right sub-trees of the URI space to specific directories), but this overriding semantic makes things more predictable.
